core/shared/platform/common/posix/posix_thread.c

```diff
@@ -47,6 +52,152 @@ os_thread_wrapper(void *arg)
     return NULL;
 }
 
+static void
+stack_context_sig_handler(int sig)
+{
+    assert(sig == SIGUSR1);
+    korp_tid self = pthread_self();
+
+    os_mutex_lock(&stack_context_handler.lock);
```

Comment:
i don't think pthread mutex is async signal safe.
if you mean to protect from a signal handler, it's more appropriate to mask the signal.